src/helpers/deserializer.test.ts

```diff
@@ -9,53 +18,95 @@ describe('Deserializer helper', () => {
-    const jsonData = {
-      id: '1',
-      type: 'TestType',
-      attributes: {
-        name: 'Name',
-        age: 25,
-      },
-    };
+    const surveyWithRelationship: Deserializer = surveyWithRelationshipFabricator();
+    const surveyWithoutRelationship: Deserializer = surveyWithoutRelationshipFabricator();
+    const questions: Deserializer[] = questionFabricator.times(1);
 
     it('returns deserialized data', () => {
-      const deserializedData = deserialize<TestType>(jsonData);
+      const deserializedData = deserialize<TestType>(testTypeFabricator());
 
       expect(deserializedData.resourceType).toBe('TestType');
-      expect(deserializedData.name).toBe('Name');
-      expect(deserializedData.age).toBe(25);
+      expect(deserializedData.name).toBe(testTypeName);
+      expect(deserializedData.age).toBe(testTypeAge);
     });
-  });
 
-  describe('deserializeList', () => {
-    const jsonArray = [
-      {
-        id: '1',
-        type: 'TestType',
-        attributes: {
-          name: 'Name',
-          age: 25,
-        },
-      },
-      {
-        id: '2',
-        type: 'TestType',
-        attributes: {
-          name: 'Name',
-          age: 30,
-        },
-      },
-    ];
-
-    it('returns a list of the deserialized items', () => {
-      const deserializedList = deserializeList<TestType>(jsonArray);
-
-      expect(deserializedList).toHaveLength(jsonArray.length);
-
-      deserializedList.forEach((item: TestType, index: number) => {
-        expect(item.resourceType).toBe('TestType');
-        expect(item.name).toEqual(jsonArray[index].attributes.name);
-        expect(item.age).toEqual(jsonArray[index].attributes.age);
+    describe('given relationships and included JSON data', () => {
+      describe('given the relationships is an array', () => {
+        it('deserializes the relation array correctly', () => {
+          const deserializedData = deserialize<Survey>(surveyWithRelationship, questions);
+
+          expect(deserializedData.resourceType).toBe('survey');
+          expect(deserializedData.id).toEqual(surveyWithRelationship.id);
+          expect(deserializedData.title).toEqual(surveyWithRelationship.attributes.title);
+          expect(deserializedData.description).toEqual(surveyWithRelationship.attributes.description);
+          expect(deserializedData.thankEmailAboveThreshold).toEqual(surveyWithRelationship.attributes.thankEmailAboveThreshold);
+          expect(deserializedData.thankEmailBelowThreshold).toEqual(surveyWithRelationship.attributes.thankEmailBelowThreshold);
+          expect(deserializedData.isActive).toEqual(surveyWithRelationship.attributes.isActive);
+          expect(deserializedData.coverImageUrl).toEqual(surveyWithRelationship.attributes.coverImageUrl);
+          expect(deserializedData.createdAt).toEqual(surveyWithRelationship.attributes.createdAt);
+          expect(deserializedData.activeAt).toEqual(surveyWithRelationship.attributes.activeAt);
```

Comment:
I think you can merge these expectations into one:
```
expect(deserializedData).toMatchObject({
    resourceType: 'survey',
    id: surveyWithRelationship.id,
    title: surveyWithRelationship.attributes.title,
    description: surveyWithRelationship.attributes.description,
    thankEmailAboveThreshold: surveyWithRelationship.attributes.thankEmailAboveThreshold,
    thankEmailBelowThreshold: surveyWithRelationship.attributes.thankEmailBelowThreshold,
    isActive: surveyWithRelationship.attributes.isActive,
    coverImageUrl: surveyWithRelationship.attributes.coverImageUrl,
    createdAt: surveyWithRelationship.attributes.createdAt,
    activeAt: surveyWithRelationship.attributes.activeAt,
  });
```
Same for others

Reply:
@hoangmirs good to know it 🤩

Reply:
@hoangmirs updated in [deb737a](https://github.com/manh-t/react-survey/pull/69/commits/deb737a0232a27e2f4994610978bb7718fcc3d7f)